### PR TITLE
Issue #13321: Kill mutation for TODOCommentCheck-1

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #13321: Kill mutation for TODOCommentCheck-1

---------

# Check
https://checkstyle.org/checks/misc/todocomment.html#TodoComment

---------

# Mutation 
https://github.com/checkstyle/checkstyle/blob/2be5780673b9ba5397526aadab981be68f874172/config/pitest-suppressions/pitest-misc-suppressions.xml#L156-L172

---------

# Explaination 
Test added

### CODE REVIEW

**Review by rdiachenko — APPROVED** [review]
lgtm

**Review by Vyom-Yadav — APPROVED** [review]
LGTM! The best type of mutation PRs!

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/todocomment/InputTodoCommentDefault.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+TodoComment
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.todocomment;
+
+public class InputTodoCommentDefault {
+    int i;
+    public void method() {
+        i++; // TODO: do differently in future // violation 'Comment matches to-do format 'TODO:''
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/todocomment/InputTodoCommentDefault.java:11]: As we targeting comment, please move `// violation ....` on line above or below and skip usage of TODO in message.
We should not mix them and we should see clearly what triggers violation.

Please keep this in mind for all comment and javadoc Checks .

**Review by romani — CHANGES_REQUESTED** [review]
Item:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/todocomment/InputTodoCommentDefault.java (hunk 1 above)
Comment (Kevin222004) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/todocomment/InputTodoCommentDefault.java]: Done

Comment (Kevin222004) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/todocomment/InputTodoCommentDefault.java]: Screenshot from 2023-07-10 11-46-18
I am not sure why this is happening.

Comment (Kevin222004) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/todocomment/InputTodoCommentDefault.java]: may be https://github.com/checkstyle/checkstyle/blob/0831953d82df93e5ebe7f178c2f2b904af8b8f3b/src/main/java/com/puppycrawl/tools/checkstyle/checks/TodoCommentCheck.java#L133 
this needs to be TokenTypes.SingleLineComment and TokenTypes.BlockComment 

I have worked with comment in fall-thru pr but i never face this in that pr

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**